### PR TITLE
BREAKING: update WithGlobalResponseType to take

### LINKED ISSUE #247: WithGlobalResponseTypes should also return XML
I would like the response type to be configurable.

schema:
```
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Example"
                }
              },
              "application/xml": {
                "schema": {
                  "$ref": "#/components/schemas/Example"
                }
              }
            },
            "description": "OK"
          },
          "400": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/HTTPError"
                }
              }
            },
            "description": "Bad Request _(validation or deserialization error)_"
          },
          "500": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/HTTPError"
                }
              }
            },
            "description": "Internal Server Error _(panics)_"
          },
```

It inserts "application/json" and it is before "application/xml", this caused it to be selected as default Content-type on documentation.

Also, all errors use "application/json" because there is no way to specify other content-type for them.
(`fuego.WithGlobalResponseTypes` do not change it)

### PR DESCRIPTION
**Pull request by dylanhitt**
Inspired by #274. 

- Changes `WithGlobalResponseType` to use `fuego.Response` as an input
- Defaults `WithGlobalResponseType` inputs to contain `application/json` and `application/xml` closes #247 
- Refactors `WithGlobalResponseType` to use the same helper as `OptionAddResponse` so we can stay in lock step with `fuego.Response` struct

Some notes: 

- I'm not the biggest fan of `openAPIResponse` for a struct name but it's just to hold the global response types until we get to generating the spec. Maybe `GlobalReponse` open to thoughts.
- thoughts on making `addResponseIfNotSet` a method

This was a fun PR. Looking forward to the feedback.

Cheers

### CODE REVIEW

File: openapi_operations_test.go
Comment (EwenQuim) [inline openapi_operations_test.go:102]: OptionAddResponse missing here!

File: openapi.go
Comment (EwenQuim) [inline openapi.go:37]: No need for it now. We can send it as a global `routeOption` that will be set at the Route creation time!

But maybe for another PR.

File: server.go
Comment (EwenQuim) [inline server.go:214]: Now it is the same as this :

```go
func WithGlobalResponseTypes(code int, description string, response Response) func(*Server) {
	return func(c *Server) {
		WithRouteOptions(
			OptionAddResponse(code, description, response),
		)(c)
	}
}
```


Or, can we remove it completely? I think it's better if we remove redundant configuration items. I think that everything that will be applied to a route must be applied as a route option.

**Review by EwenQuim — CHANGES_REQUESTED** [review]
This was a huge PR. I think it's really good! 

I think we can do further refacto but maybe you can do it in another PR.

Sorry for the 3 days delay!

I mark as "request changes" for the missing `OptionAddError` used in tests.

File: server.go
Comment (dylanhitt) [reply server.go:214]: Hmm let met test this real quick. 

I personally really like being allowed to set global response types for servers/groups.

Comment (dylanhitt) [reply server.go:214]: Yeah these two things aren't the same.

```
--- FAIL: TestWithGlobalResponseType (0.00s)
    --- FAIL: TestWithGlobalResponseType/errors_with_route_overrides (0.00s)
        openapi_operations_test.go:104: 
                Error Trace:    /Users/dhitt/github/go-fuego/fuego/openapi_operations_test.go:104
                Error:          Not equal: 
                                expected: "My Global Error"
                                actual  : "Bad Request _(validation or deserialization error)_"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -My Global Error
                                +Bad Request _(validation or deserialization error)_
                Test:           TestWithGlobalResponseType/errors_with_route_overrides
                Messages:       Overrides Fuego's default 400 error
```

Comment (dylanhitt) [reply server.go:214]: I guess we can just use WithRouteOptions here as well. I forget that exists always. Let me figure out the reason for the error above and I'll submit another PR

Comment (dylanhitt) [reply server.go:214]: At least I tested this well enough 😅

Comment (dylanhitt) [reply server.go]: Okay, nice catch. Just needed to change up this: https://github.com/go-fuego/fuego/pull/276/commits/81a7afcda321dc40268a9f1b0d08a7d7d7dc2feb

So couple things. Yes we should deprecate `WithGlobalResponseType`. I think we should get this change merge with a deprecation notice and best way to migrate to `WithRouteOptions`. Emphasis on maintaining the two default api responses declare as well. I'll probably move those over to use `WithRouteOptions` in this MR as well so we don't forget. 

How does that sound?

File: openapi.go
Comment (dylanhitt) [reply openapi.go:37]: Assuming you agree with https://github.com/go-fuego/fuego/pull/276#discussion_r1887167481

This will just be removed after deprecation after the deprecation notice of this thing on the next release.

File: server.go
Comment (EwenQuim) [reply server.go]: I think it looks good!

And yes moving back the default server options is really nice as well, that's something we wanted to do anyway.

**Comment by EwenQuim** [comment]
I'll wait for your PR to be finished before merging mine, because there will be conflicts I think ;) 

Only the deprecation notice for `WithGlobalResponseType` is missing. And I think you can use WithRouteOptions in  `NewServer` in `server.go` file.

File: server.go
Comment (dylanhitt) [reply server.go]: Okay cool, I’ll get this don’t either tonight or tomorrow night then.

**Comment by dylanhitt** [comment]
Updated

**Review by EwenQuim — APPROVED** [review]

**Review by ccoVeille — COMMENTED** [review]
LGTM :+1: 

But I won't approve as the changes are out of my knowlege in term of regression.

**Pull request merged**
